### PR TITLE
[Darkside] Search CSS update

### PR DESCRIPTION
**Pull request by KenAJoh**

### CODE REVIEW

File: @navikt/core/react/src/form/search/Search.tsx [hunk 1]
```diff
@@ -142,6 +146,33 @@ export const Search = forwardRef<HTMLInputElement, SearchProps>(
...
+      clearButton && !inputProps.disabled && (value ?? internalValue);
+
+    const ClearButton = () =>
+      themeContext ? (
+        <Button
+          className="navds-search__button-clear"
+          variant="tertiary-neutral"
+          size={size === "medium" ? "small" : "xsmall"}
+          icon={<XMarkIcon aria-hidden />}
+          aria-label={clearButtonLabel || translate("clear")}
```
Comment (KenAJoh) [inline @navikt/core/react/src/form/search/Search.tsx]:
```suggestion
          title={clearButtonLabel || translate("clear")}
```

**Comment by HalvorHaugan** [comment]
Please resolve conflicts.

**Comment by KenAJoh** [comment]
TODO: Set placeholder to `text-neutral`

File: @navikt/core/css/darkside/form/search.darkside.css [hunk 2]
```diff
@@ -25,204 +9,140 @@
...
-  .navds-search__button-search.navds-button--secondary:focus:active {
-    box-shadow:
-      0 0 0 1px var(--ac-search-button-focus-active-border, var(--a-surface-default)) inset,
-      var(--a-shadow-focus);
+/* Special-case where we hover/focus input, and want the secondary-button to match input-border */
+.navds-search:not(.navds-search--error, .navds-search--disabled) {
+  & .navds-search__wrapper:has(.navds-search__input:is(:hover, :focus-visible)) {
+    & .navds-search__button-search.navds-button--secondary:not(:hover, :active) {
+      --__axc-button-border-color: var(--ax-border-accent-strong);
+    }
```
Comment (HalvorHaugan) [inline @navikt/core/css/darkside/form/search.darkside.css]: Why use nesting here?

File: @navikt/core/css/darkside/form/search.darkside.css [hunk 3]
```diff
@@ -25,204 +9,140 @@
...
+  box-shadow: inset 0 0 0 1px var(--ax-border-danger);
 
-.navds-search--error .navds-search__input:focus-visible:not(:hover, :disabled) {
-  box-shadow:
-    inset 0 0 0 1px var(--ac-search-error-border, var(--__ac-search-error-border, var(--a-border-danger))),
-    var(--a-shadow-focus);
+  &:not(.navds-search__input--simple) {
+    box-shadow:
+      inset -2px 0 0 0 var(--ax-border-danger),
+      inset 0 0 0 1px var(--ax-border-danger);
```
Comment (HalvorHaugan) [inline @navikt/core/css/darkside/form/search.darkside.css]: Wouldn't it be easier to flip this?
```css
  box-shadow:
    inset -2px 0 0 0 var(--ax-border-danger),
    inset 0 0 0 1px var(--ax-border-danger);

  &.navds-search__input--simple {
    box-shadow: inset 0 0 0 1px var(--ax-border-danger);
  }
}
```

File: @navikt/core/css/darkside/form/search.darkside.css [hunk 4]
```diff
@@ -25,204 +9,140 @@
...
   }
 }
 
-/* Error-handling */
-.navds-search--error .navds-search__input:not(:hover, :disabled) {
-  border-color: var(--ac-search-error-border, var(--__ac-search-error-border, var(--a-border-danger)));
-  box-shadow: inset 0 0 0 1px var(--ac-search-error-border, var(--__ac-search-error-border, var(--a-border-danger)));
-}
+/* --------------------------- Search Error-state --------------------------- */
+.navds-search--error .navds-search__input:not(:disabled) {
```
Comment (HalvorHaugan) [inline @navikt/core/css/darkside/form/search.darkside.css:125]: Should we have a story for disabled state, and one for disabled+error?

File: @navikt/core/react/src/form/search/Search.tsx [hunk 5]
```diff
@@ -142,6 +146,33 @@ export const Search = forwardRef<HTMLInputElement, SearchProps>(
...
+    const ClearButton = () =>
+      themeContext ? (
+        <Button
+          className="navds-search__button-clear"
+          variant="tertiary-neutral"
+          size={size === "medium" ? "small" : "xsmall"}
+          icon={<XMarkIcon aria-hidden />}
+          title={clearButtonLabel || translate("clear")}
+          hidden={!showClearButton}
+        />
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/form/search/Search.tsx]:
```suggestion
          hidden={!showClearButton}
          onClick={(event) => handleClear({ trigger: "Click", event })}
        />
```

File: @navikt/core/css/darkside/form/search.darkside.css (hunk 2 above)
Comment (KenAJoh) [reply @navikt/core/css/darkside/form/search.darkside.css]: Mostly for readability, but looks like prettier makes it take one line each anyways so removed the nesting 👍

File: @navikt/core/css/darkside/form/search.darkside.css (hunk 3 above)
Comment (KenAJoh) [reply @navikt/core/css/darkside/form/search.darkside.css]: Good call 👍

File: @navikt/core/css/darkside/form/search.darkside.css
Comment (HalvorHaugan) [inline @navikt/core/css/darkside/form/search.darkside.css:1]: Button has hover effect when disabled in forced-colors mode
image
Also, the contrast between button background and icon on primary is not ideal.

Comment (KenAJoh) [reply @navikt/core/css/darkside/form/search.darkside.css:1]: Nice catch, this is a problem with "Button" it seems. Will create a separate fix for it 👍

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**